Formula/ns-3.rb

```diff
@@ -16,53 +15,86 @@ class Ns3 < Formula
   end
 
   depends_on "boost" => :build
-  depends_on "python@3.10" => [:build, :test]
+  depends_on "cmake" => :build
+  depends_on xcode: [:build, "11"]
+
+  depends_on "gsl"
+  depends_on "open-mpi"
+  depends_on "python@3.10"
 
-  uses_from_macos "libxcrypt"
   uses_from_macos "libxml2"
   uses_from_macos "sqlite"
 
+  # Clears the Python3_LIBRARIES variable. Removing `PRIVATE ${Python3_LIBRARIES}`
+  # in ns3-module-macros is not sufficient as it doesn't apply to visualizer.so.
+  on_macos do
+    patch :DATA
+  end
+
   on_linux do
     depends_on "gcc"
   end
 
-  # `gcc version 5.4.0 older than minimum supported version 7.0.0`
+  # Needs GCC 8 or above
   fails_with gcc: "5"
+  fails_with gcc: "6"
+  fails_with gcc: "7"
 
   resource "pybindgen" do
     url "https://files.pythonhosted.org/packages/e0/8e/de441f26282eb869ac987c9a291af7e3773d93ffdb8e4add664b392ea439/PyBindGen-0.22.1.tar.gz"
     sha256 "8c7f22391a49a84518f5a2ad06e3a5b1e839d10e34da7631519c8a28fcba3764"
   end
 
-  # Fix ../src/lte/model/fdmt-ff-mac-scheduler.cc:1044:16: error:
-  # variable 'bytesTxed' set but not used [-Werror,-Wunused-but-set-variable]
-  # TODO: remove in the next release.
-  patch do
-    url "https://gitlab.com/nsnam/ns-3-dev/-/commit/dbd49741fcd5938edec17eddcde251b5dee25d05.diff"
-    sha256 "28e92297cfe058cfb587527dc3cfdb8ac66b51aba672be29539b6544591e2f1e"
-  end
-
   def install
-    resource("pybindgen").stage buildpath/"pybindgen"
+    resource("pybindgen").stage buildpath/"../pybindgen"
+    ENV.append "PYTHONPATH", "#{buildpath}/../pybindgen"
```

Comment:
```suggestion
    resource("pybindgen").stage buildpath.parent/"pybindgen"
    ENV.append "PYTHONPATH", buildpath.parent/"pybindgen"
```
Can we not put this inside `buildpath`?

Reply:
The `ns3` wrapper expected it to be in the parent folder, but since we call cmake directly, it is no longer necessary.